Xamarin.Forms.Core/BindingExpression.cs

```diff
@@ -727,14 +727,20 @@ public void PropertyChanged(object sender, PropertyChangedEventArgs args)
 					}
 				}
 
-				if (Device.IsInvokeRequired)
+				Action action = () => _expression.Apply();
+				if (_expression._weakTarget.TryGetTarget(out BindableObject obj) && obj.Dispatcher != null && obj.Dispatcher.IsInvokeRequired)
```

Comment:
we might have to check how this behaves with compiled bindings (and't recall if they use this code path or not)